app/src/test/java/org/mozilla/fenix/helpers/FenixRobolectricTestRunner.kt

```diff
@@ -0,0 +1,31 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.helpers
+
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+
+/**
+ * A test runner that starts Robolectric with our custom configuration for use in unit tests.
+ *
+ * usage:
+ * ```
+ * @RunWith(FenixRobolectricTestRunner::class)
+ * class ExampleUnitTest {
+ * ```
+ *
+ * IMPORTANT NOTES:
+ * - This should ALWAYS be used instead of RobolectricTestRunner and AndroidJUnit4 (note: the latter
+ * just delegates to the former)
+ * - You should only use Robolectric when necessary because it non-trivially increases test duration.
+ */
+class FenixRobolectricTestRunner(testClass: Class<*>) : RobolectricTestRunner(testClass) {
```

Comment:
I find the name of this to be a little bit confusing because this test runner replace both the @RunWith(RobolectricTestRunner::class) and @RunWith(AndroidJUnit4::class). Someone may not realize that AndroidJUnit4 is also replaced by this when writing a future test and may opt to use AndroidJUnit4 instead of your test runner.

Reply:
> Someone may not realize that AndroidJUnit4 is also replaced by this when writing a future test and may opt to use AndroidJUnit4 instead of your test runner.

My hope is that the linter would catch that. Unfortunately, I don't think I can name it in a way that captures the intent of both `AndroidJUnit4` and `FenixRobolectricTestRunner`. :\

In any case, as we discussed, I can add a comment to explain why `AndroidJUnit4` delegates to `FenixRobolectricTestRunner`

Reply:
That would be great!

Reply:
New class comment:
```kotlin
/**
 * A test runner that starts Robolectric with our custom configuration for use in unit tests. This
 * should ALWAYS be used instead of RobolectricTestRunner and AndroidJUnit4. You should only use
 * Robolectric when necessary because it non-trivially increases test duration.
 *
 * usage:
 * ```
 * @RunWith(FenixRobolectricTestRunner::class)
 * class ExampleUnitTest {
 * ```
 *
 * There were three common test runners before this patch:
 * 1. The default (@RunWith not specified) = JUnit4
 * 2. @RunWith(RobolectricTestRunner::class) = JUnit4 with support for the Android framework via Robolectric
 * 3. @RunWith(AndroidJUnit4::class) = JUnit4 with support for the Android framework. This currently
 * delegates to Robolectric but is presumably generically named so that it can support different
 * implementations in the future. The name creates confusion on over the difference between this and
 * JUnit without any Android support (1).
 *
 * We chose the name RobolectricTestRunner because we want folks to know they're starting Robolectric
 * because it increases test runtime. Furthermore, the naming of 3) is unclear so we didn't want to
 * use that name.
 */
```